### PR TITLE
Handle when the R process crashes

### PR DESCRIPTION
**Pull request by gaborcsardi**
On top of #61 

* In interactive sessions, we need to reset some signal handlers, we do this from the startup profile.  This is a call to processx.
* While in the profile, we also make the inherited handles non-inheritable, with another processx call.
* `$read()` returns 500, 501 or 502, depending or why/how the process finished. 500 is normal termination (exit code 0), 501 is a crash (non-zero exit code), 502 when the process channel is closed, but the process is alive.
* After an 502, we kill the process.

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -80,6 +80,6 @@ is_complete_expression <- function(x) {
 
 bold <- function(x) {
   tryCatch(
-    utils::getFromNamespace("bold", "crayon")(x),
```
Comment (jimhester) [inline R/utils.R]: Why this change, do you plan to stop exporting bold from crayon in the future?

File: R/check.R [hunk 2]
```diff
@@ -24,7 +24,7 @@ convert_and_check_my_args <- function(options) {
   ## Conversions
   options <- within(options, {
     if (has("libpath")) libpath <- as.character(libpath)
-    if (has("repos")) stopifnot(is.character(repos))
+    if (has("repos")) repos <- repos
```
Comment (jennybc) [inline R/check.R]: Is this reversion intentional?

Comment (gaborcsardi) [reply R/check.R]: Yeah, I decided that we just pass is as it is. The child process will error if it is not good. I guess I could also  just remove that line completely.

**Comment by gaborcsardi** [comment]
OK, this looks good now.

**Pull request merged**